stake-pool/cli/src/main.rs

```diff
@@ -1611,6 +1856,61 @@ fn main() {
                           Defaults to the token-owner's associated pool token account. \
                           Creates the account if it does not exist."),
             )
+            .arg(
+                Arg::with_name("referrer")
+                    .validator(is_pubkey)
+                    .value_name("ADDRESS")
+                    .takes_value(true)
+                    .help("Account to receive the referral fees for deposits. \
+                          Defaults to the pool manager."),
+            )
+        )
+        .subcommand(SubCommand::with_name("deposit-sol")
+            .about("Deposit SOL into the stake pool in exchange for pool tokens")
+            .arg(
+                Arg::with_name("pool")
+                    .index(1)
+                    .validator(is_pubkey)
+                    .value_name("POOL_ADDRESS")
+                    .takes_value(true)
+                    .required(true)
+                    .help("Stake pool address"),
+            ).arg(
+                Arg::with_name("amount")
+                    .index(2)
+                    .validator(is_amount)
+                    .value_name("AMOUNT")
+                    .takes_value(true)
+                    .help("Amount in SOL to deposit into the stake pool reserve account."),
+            )
+            .arg(
+                Arg::with_name("from")
+                    .long("from")
+                    .validator(is_keypair)
+                    .value_name("KEYPAIR")
+                    .takes_value(true)
+                    .help("Account to deposit SOL from. \
+                          Defaults to the fee payer."),
+            )
+            .arg(
+                Arg::with_name("token_receiver")
+                    .long("token-receiver")
+                    .validator(is_pubkey)
+                    .value_name("POOL_TOKEN_RECEIVER_ADDRESS")
+                    .takes_value(true)
+                    .help("Account to receive the minted pool tokens. \
+                          Defaults to the token-owner's associated pool token account. \
+                          Creates the account if it does not exist."),
+            )
+            .arg(
+                Arg::with_name("referrer")
+                    .long("referrer")
+                    .validator(is_pubkey)
+                    .value_name("REFERRER_TOKEN_ADDRESS")
+                    .takes_value(true)
+                    .help("Account to receive the referral fees for deposits. \
+                          Defaults to the pool manager."),
```

Comment:
Just thinking, if someone is using this from the CLI, how about they get the referral instead?
```suggestion
                    .help("Pool token account to receive the referral fees for deposits. \
                          Defaults to the token receiver."),
```